docs/sources/developers/plugins/extend-the-grafana-ui-with-links.md

```diff
@@ -1,25 +1,78 @@
 ---
-title: Use extensions to add links to app plugins
+title: Use extensions to add new functionality in the Grafana user interface
 description: Learn how to add links to the Grafana user interface from an app plugin
 ---
 
-# Use extensions to add links to app plugins
+# The when, why and how of UI extensions
 
-You can use the Plugin extensions API with your Grafana app plugins to add links to the Grafana UI. This feature lets you send users to your plugin's pages from other spots in the Grafana application.
+Before we go into details we need to cover the two major concepts in the UI extensions feature of Grafana.
+
+1. Extension point - a place in the UI where plugins can provide new functionality to the end user. Examples of extension points are the dashboard panel menu or toolbar in explore.
```

Comment:
I think we need to make it clearer that Grafana provides some extension points, currently the dashboard panel menu or toolbar in explore, which app plugin developers can hook into; and that App plugins can also provide additional extension points. How about: 

```suggestion
1. Extension point - a place in the UI where plugins can provide new functionality to the end user. The Grafana UI currently exposes extension points in the panel menu within a dashboard, and the toolbar in Explore. App plugins can also define their own extension points.
```